apps/src/lib/wallet/pre_genesis.rs

```diff
@@ -36,70 +36,72 @@ pub fn gen_and_store(
     let wallet_dir = wallet_path.parent().unwrap();
     fs::create_dir_all(wallet_dir)?;
     // Write the file
-    let options = FileOptions::new().create(true).write(true).truncate(true);
-    let mut filelock =
-        FileLock::lock(wallet_path.to_str().unwrap(), true, options)?;
-    filelock.file.write_all(&data)?;
+    let mut options = fs::OpenOptions::new();
+    options.create(true).write(true).truncate(true);
+    let mut lock = RwLock::new(options.open(wallet_path)?);
+    let mut guard = lock.write()?;
+    guard.write_all(&data)?;
     Ok(validator)
 }
 
 /// Try to load and decrypt keys, if encrypted, in a [`ValidatorWallet`]
 /// from a TOML file.
 pub fn load(store_dir: &Path) -> Result<ValidatorWallet, ReadError> {
     let wallet_file = validator_file_name(store_dir);
-    match FileLock::lock(
-        wallet_file.to_str().unwrap(),
-        true,
-        FileOptions::new().read(true).write(false),
-    ) {
-        Ok(mut filelock) => {
-            let mut store = Vec::<u8>::new();
-            filelock.file.read_to_end(&mut store).map_err(|err| {
-                ReadError::ReadWallet(
-                    store_dir.to_str().unwrap().into(),
-                    err.to_string(),
-                )
-            })?;
-            let store =
-                ValidatorStore::decode(store).map_err(ReadError::Decode)?;
+    let mut options = fs::OpenOptions::new();
+    options.read(true).write(false);
+    let lock = RwLock::new(options.open(&wallet_file).map_err(|err| {
+        ReadError::ReadWallet(
+            wallet_file.to_string_lossy().into_owned(),
+            err.to_string(),
+        )
+    })?);
+    let guard = lock.read().map_err(|err| {
+        ReadError::ReadWallet(
+            wallet_file.to_string_lossy().into_owned(),
+            err.to_string(),
+        )
+    })?;
+    let mut store = Vec::<u8>::new();
+    (&*guard).read_to_end(&mut store).map_err(|err| {
```

Comment:
hm, the compiler isn't able to figure out the proper deref here (`&*guard`)?

Reply:
Unfortunately not because `read_to_end` takes `&mut self`, but the `trait Read` is implemented for `&File`. I think it's probably trying to auto-deref as `&mut self` which it can't